test/integration/arbitrum-mainnet/Loans.fork.t.sol

```diff
@@ -463,21 +482,160 @@ contract LoansForkTest is LoansTestBase {
         assertGt(protocolFee, 0);
     }
 
-    function closeAndCheckLoan(uint loanId, uint loanAmount, uint lateFee) internal {
+    function closeAndCheckLoan(uint loanId, uint loanAmount, uint totalAmountToSwap, uint lateFee) internal {
         // Track balances before closing
         uint userCashBefore = pair.cashAsset.balanceOf(user);
         uint userUnderlyingBefore = pair.underlying.balanceOf(user);
+        uint expiration = pair.takerNFT.getPosition(loanId).expiration;
+        // Get current price from oracle
+        (uint currentPrice,) = pair.oracle.pastPriceWithFallback(uint32(expiration));
+        // Convert cash amount to expected underlying amount using oracle's conversion
+        uint expectedUnderlying = pair.oracle.convertToBaseAmount(totalAmountToSwap, currentPrice);
 
-        // no price change so underlying out should be underlying in minus slippage and minus late fees
-        uint minUnderlyingOut = underlyingAmount;
-        uint minUnderlyingOutWithSlippage = (minUnderlyingOut * (100 - slippage) / 100) - lateFee;
+        uint minUnderlyingOutWithSlippage = expectedUnderlying * (100 - slippage) / 100 - lateFee;
         uint underlyingOut = closeLoan(pair, user, loanId, minUnderlyingOutWithSlippage);
+
         assertGe(underlyingOut, minUnderlyingOutWithSlippage);
```

Comment:
This check should use tolerance and compare both ways. Right now, if it's using `totalAmountToSwap` that is too small, the check is ineffective. In most cases above, the `totalAmountToSwap` is actually too small (since it's passed as `loanAmount` but should also include position withdrawal. So this issue is "masked" by the one-directional check.